### PR TITLE
VR-8155 delete bindings for repositories during migration

### PR DESCRIPTION
**Pull request by evgeniilezhnin**

### CODE REVIEW

File: backend/src/main/java/ai/verta/modeldb/batchProcess/CollaboratorResourceMigration.java [hunk 1]
```diff
@@ -297,6 +301,7 @@ private static void migrateRepositories() {
               }
             }
           }
+          deleteRoleBindingsOfRepositories(migratedRepositoryEntities);
```
Comment (conradoverta) [inline backend/src/main/java/ai/verta/modeldb/batchProcess/CollaboratorResourceMigration.java]: This needs to be inside the migration step. We cannot set `repository.setVisibility_migration(true);` and then have this. It's possible to have a failure between these steps and the role binding will never be removed, leading to inconsistencies. We need to make our migrations safe.

**Review by conradoverta — CHANGES_REQUESTED** [review]
We also need to do this for projects, right?

File: backend/src/main/java/ai/verta/modeldb/batchProcess/CollaboratorResourceMigration.java (hunk 1 above)
Comment (evgeniilezhnin) [reply backend/src/main/java/ai/verta/modeldb/batchProcess/CollaboratorResourceMigration.java]: updated

**Comment by evgeniilezhnin** [comment]
> We also need to do this for projects, right?

It was done here: https://github.com/VertaAI/modeldb/pull/1870/files

Here is a fix after your recent comment: https://github.com/VertaAI/modeldb/pull/1876

File: backend/src/main/java/ai/verta/modeldb/batchProcess/CollaboratorResourceMigration.java (hunk 1 above)
Comment (conradoverta) [reply backend/src/main/java/ai/verta/modeldb/batchProcess/CollaboratorResourceMigration.java]: This is not done before `repository.setVisibility_migration(true);`. The code has the same issue as before: I can mark the migration as completed before it's completed. We can only set the migration as done if every single operation we'll do for that given element is done and successful. If they are not, we *cannot* set the migration as done.

Comment (evgeniilezhnin) [reply backend/src/main/java/ai/verta/modeldb/batchProcess/CollaboratorResourceMigration.java]: Sorry, my bad. Updated again

**Review by conradoverta — APPROVED** [review]

File: backend/src/main/java/ai/verta/modeldb/batchProcess/CollaboratorResourceMigration.java (hunk 1 above)
Comment (conradoverta) [reply backend/src/main/java/ai/verta/modeldb/batchProcess/CollaboratorResourceMigration.java]: Thanks! It's a small change, but it's good for us to develop this muscle memory.

**Pull request merged**